x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java

```diff
@@ -617,6 +619,11 @@ private static Settings getTokenIndexSettings() {
             .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1")
             .put(IndexMetadata.SETTING_PRIORITY, 1000)
             .put(IndexMetadata.INDEX_FORMAT_SETTING.getKey(), INTERNAL_TOKENS_INDEX_FORMAT)
+            // since {@code TokenService#VERSION_GET_TOKEN_DOC_FOR_REFRESH}
+            // For tokens, we issue explicit refreshes before searches rather than the more common refresh-after-write strategy.
+            // Token docs are infrequently searched and the extra latency due to the possibly throttled refresh is tolerable.
+            // Consequently, no automatic periodic refresh is necessary.
+            .put("index.refresh_interval", "-1")
```

Comment:
I don't see a real need to disable the periodic refresh. IIUC, it does not count towards the refresh throttling. Previously we had refresh-after-write. In theory it could also allow the periodic refresh similar to today's refresh-before-search. So I don't see this being a strong reason to disable periodic refresh. Also periodic refresh only happens if the index has received search requests within last 30 seconds. If searches are rare enough, having the default `refresh_interval` behaves similar to "no-periodic-refresh" most of the time. Disabling periodic refresh could also be trappy when someone needs to interogate the token index directly.

Reply:
Thats good input, Yang!
I agree, there is not a strong case to disabling tokens index automatic refreshes.
The savings in the number of refresh calls in not big (if at all).

I think the tradeoffs are between:
 1) possibly fewer overall refreshes, only an explicit one when the token cleanup job is kicked off, at most every 30 mins by every node handling token invalidation requests
 2) possibly more refreshes, because of the extra automatic refreshes following the cleanup delete-by-query job, but these refreshes don't count towards the refresh budget

I'm leaning to not disable automatic refreshes, as you say, on grounds that it can be trappy to remember to always refresh before searching the tokens index to get results with any reasonable recency.
Thoughts @slobodanadamovic @jakelandis ?

Reply:
I've reverted the token index changes.

Reply:
I am pretty neutral. No concerns with leaving this as the default, and no concerns if we disable it. However, disabling is probably a micro optimization and agree in some specific cases can be trappy.